GameSentenceMiner/util/cron/jiten_update.py

```diff
@@ -0,0 +1,424 @@
+            update_fields['title_romaji'] = jiten_data['title_romaji']
+        elif 'title_romaji' in manual_overrides:
+            skipped_fields.append('title_romaji')
+        
+        if 'title_english' not in manual_overrides and jiten_data.get('title_english'):
+            update_fields['title_english'] = jiten_data['title_english']
+        elif 'title_english' in manual_overrides:
+            skipped_fields.append('title_english')
+        
+        if 'type' not in manual_overrides and jiten_data.get('media_type'):
+            # Map media type to string
+            media_type_map = {1: 'Anime', 7: 'Visual Novel', 2: 'Manga'}
+            update_fields['game_type'] = media_type_map.get(jiten_data['media_type'], 'Unknown')
+        elif 'type' in manual_overrides:
+            skipped_fields.append('type')
+        
+        if 'description' not in manual_overrides and jiten_data.get('description'):
+            update_fields['description'] = jiten_data['description']
+        elif 'description' in manual_overrides:
+            skipped_fields.append('description')
+        
+        if 'difficulty' not in manual_overrides and jiten_data.get('difficulty') is not None:
+            update_fields['difficulty'] = jiten_data['difficulty']
+        elif 'difficulty' in manual_overrides:
+            skipped_fields.append('difficulty')
+        
+        if 'character_count' not in manual_overrides and jiten_data.get('character_count') is not None:
+            update_fields['character_count'] = jiten_data['character_count']
+        elif 'character_count' in manual_overrides:
+            skipped_fields.append('character_count')
+        
+        if 'links' not in manual_overrides and jiten_data.get('links'):
+            update_fields['links'] = jiten_data['links']
+        elif 'links' in manual_overrides:
+            skipped_fields.append('links')
+        
+        if 'release_date' not in manual_overrides and jiten_data.get('release_date'):
+            update_fields['release_date'] = jiten_data['release_date']
+        elif 'release_date' in manual_overrides:
+            skipped_fields.append('release_date')
+        
+        # Always re-download image if not manually overridden
+        if 'image' not in manual_overrides and jiten_data.get('cover_name'):
+            try:
+                logger.debug(f"🖼️ Downloading image: {jiten_data['cover_name']}")
+                img_response = requests.get(jiten_data['cover_name'], timeout=10)
+                if img_response.status_code == 200:
+                    # Encode image to base64
+                    img_base64 = base64.b64encode(img_response.content).decode('utf-8')
+                    
+                    # Detect image format from content-type header or magic bytes
+                    content_type = img_response.headers.get('content-type', '').lower()
+                    if 'png' in content_type:
```

Comment:
does jiten api gurantee file type?

Reply:
yes, guaranteed to be jpg https://github.com/Sirush/Jiten/blob/bc2e017f63625918eea1ef4523207cea06e67a11/Jiten.Core/JitenHelper.cs#L25